packages/web3-shared/src/hooks/useNativeTokenDetailed.ts

```diff
@@ -1,21 +1,18 @@
 import { useAsyncRetry } from 'react-use'
-import { useTokenConstants } from '../constants'
-import { EthereumTokenType, NativeTokenDetailed } from '../types'
-import { getChainDetailed } from '../utils'
+import type { NativeTokenDetailed, NativeToken, FungibleTokenDetailed } from '../types'
+import { createNativeToken } from '../utils'
 import { useChainId } from './useChainId'
 
 export function useNativeTokenDetailed() {
     const chainId = useChainId()
-    const { NATIVE_TOKEN_ADDRESS } = useTokenConstants()
-    return useAsyncRetry(async (): Promise<NativeTokenDetailed> => {
-        const nativeCurrency = getChainDetailed(chainId)?.nativeCurrency
-        return {
-            type: EthereumTokenType.Native,
-            address: NATIVE_TOKEN_ADDRESS,
-            chainId,
-            name: nativeCurrency?.name ?? 'Unknown',
-            symbol: nativeCurrency?.symbol ?? '',
-            decimals: nativeCurrency?.decimals ?? 0,
-        }
-    }, [chainId, NATIVE_TOKEN_ADDRESS])
+
+    return useAsyncRetry(async (): Promise<NativeTokenDetailed> => createNativeToken(chainId), [chainId])
+}
+
+export function useNativeTokensDetailed(listOfToken: Pick<NativeToken, 'address' | 'type'>[]) {
+    const chainId = useChainId()
+
+    return useAsyncRetry(async (): Promise<FungibleTokenDetailed[]> => {
+        return listOfToken.map(() => createNativeToken(chainId))
+    }, [chainId])
```

Comment:
Do you forget to add the `listOfToken` into the dependence list?

Reply:
done